### PR TITLE
lib: at_cmd_parser: Update documentation

### PR DESCRIPTION
**Pull request by metc**
Documentation update of the module header file.

Signed-off-by: Christopher Métrailler <christopher.metrailler@nordicsemi.no>

### CODE REVIEW

**Comment by metc** [comment]
@evenl FYI

**Comment by ru-fu** [comment]
No TW review is required if the update only affects the API documentation.
You should consider though if any changes are needed to the conceptual documentation (the rst file).

**Comment by metc** [comment]
> No TW review is required if the update only affects the API documentation.
> You should consider though if any changes are needed to the conceptual documentation (the rst file).

Thank you for this note @ru-fu. I updated the file `at_cmd_parser.rst `. Could you please have a look?

File: include/at_cmd_parser/at_cmd_parser.rst [hunk 1]
```diff
@@ -3,13 +3,13 @@
 AT command parser
 #################
 
-The AT command parser is a library for parsing the string that is returned after issuing an AT command.
-It parses the response parameters from the string and saves them in a list.
+The AT command parser is a library for parsing either a string response returned after issuing an AT command, an unsolicited event or a notification.
```
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```Suggestion
The AT command parser is a library which parses any of the following:

- A string response returned after issuing an AT command
- An unsolicited event
- A notification
````

File: include/at_cmd_parser/at_cmd_parser.rst [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
+It parses the response, notification or event, extract the parameters from the string and saves them in a structured list.
```
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```Suggestion
After parsing the response, notification, or event, the AT command parser extracts the parameters from the string and saves them in a structured list.

````

File: include/at_cmd_parser/at_cmd_parser.rst [hunk 3]
```diff
@@ -3,13 +3,13 @@
...
 
-To store the parameters in a list, the parser uses the :ref:`at_params_readme` storage module.
+The :ref:`at_params_readme` storage module is used to store the parameters. Each element in the list is a parameter, identified by a type, a length and a value.
```
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```Suggestion
The :ref:`at_params_readme` storage module is used to store the parameters. Each element in the list is a parameter, identified by a specific type, length, and value.
```

File: include/at_cmd_parser/at_cmd_parser.rst [hunk 4]
```diff
@@ -3,13 +3,13 @@
...
 The resulting list can be used by any external module to obtain the value of the parameters.
-Based on the type of the parameter, you can obtain its value by calling either
-:cpp:func:`at_params_int_get`, :cpp:func:`at_params_short_get`, :cpp:func:`at_params_array_get`, or :cpp:func:`at_params_string_get`.
+Based on the type of the parameter, you can obtain its value by calling either :cpp:func:`at_params_int_get`, :cpp:func:`at_params_short_get`, :cpp:func:` at_params_array_get`, or :cpp:func:`at_params_string_get`.
```
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```Suggestion
Based on the type of the parameter, you can obtain its value by calling :cpp:func:`at_params_int_get`, :cpp:func:`at_params_short_get`, :cpp:func:` at_params_array_get`, or :cpp:func:`at_params_string_get`.

File: include/at_cmd_parser/at_cmd_parser.rst [hunk 5]
```diff
@@ -3,13 +3,13 @@
...
+The function :cpp:func:`at_params_size_get` can be used to read the length of a parameter or to check if a parameter exists at a specified index.
```
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```Suggestion
The function :cpp:func:`at_params_size_get` can be used to read the length of a parameter or to check if a parameter exists at a specified index of the list.

````

File: include/at_cmd_parser/at_cmd_parser.rst (hunk 3 above)
Comment (umapraseeda) [inline include/at_cmd_parser/at_cmd_parser.rst]:
```` 
Suggestion: Please start the sentence: "Each element in the list.. " in a new line. 
````

**Comment by metc** [comment]
@umapraseeda Thank you for your comments and suggestions. I think I have addressed all of them.

**Comment by umapraseeda** [comment]
> @umapraseeda Thank you for your comments and suggestions. I think I have addressed all of them.

Thank you.

**Pull request merged**